SpatialGDK/Source/SpatialGDK/Public/SpatialView/CommandRequest.h

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Templates/UniquePtr.h"
+#include <improbable/c_worker.h>
+#include <improbable/c_schema.h>
+
+namespace SpatialGDK
+{
+
+struct CommandRequestDeleter
+{
+	void operator()(Schema_CommandRequest* CommandRequest) const noexcept
+	{
+		if (CommandRequest)
+		{
+			Schema_DestroyCommandRequest(CommandRequest);
+		}
+	}
+};
+
+using OwningCommandRequestPtr = TUniquePtr<Schema_CommandRequest, CommandRequestDeleter>;
+
+// An RAII wrapper for component data.
+class CommandRequest
+{
+public:
+	// Creates a new component data.
+	explicit CommandRequest(Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
+	// Takes ownership of component data.
+	explicit CommandRequest(OwningCommandRequestPtr Data, Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
+
+	~CommandRequest() = default;
+
+	// Moveable, not copyable.
+	CommandRequest(const CommandRequest&) = delete;
+	CommandRequest(CommandRequest&&) = default;
+	CommandRequest& operator=(const CommandRequest&) = delete;
+	CommandRequest& operator=(CommandRequest&&) = default;
+
+	static CommandRequest CreateCopy(const Schema_CommandRequest* Data, Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
```

Comment:
Perhaps this should be a constructor from taking const Schema_CommandRequest* as an argument?

Not sure if CreateCopy is the correct name for this function.

Reply:
A constructor taking a `const Schema_CommandRequest*` is what I use in my own stuff but there were concerns this might be confusing. I personally think it's fine but this was to address that.

It is creating a copy of whatever data you are giving to it.